cmd/check_run.go

```diff
@@ -0,0 +1,87 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/engine"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/formatters"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var checkRunCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Run a single check inside of podman unshare",
+	Long: `This command will run a check while wrapped inside of podman unshare.
+It is a hidden command, and is not meant to be called by the user.
+It takes its input from environment variables only.`,
+	Hidden: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Expect an environment variable named PREFLIGHT_CHECK_EXEC
+		// It will specify the one check to execute inside of the podman unshare
+		// environment.
+		enabledCheck, ok := os.LookupEnv("PREFLIGHT_EXEC_CHECK")
+		if !ok {
+			log.Fatal("Enabled check envvar not specified")
+			os.Exit(1)
+		}
+		image, ok := os.LookupEnv("PREFLIGHT_EXEC_IMAGE")
+		if !ok {
+			log.Fatal("Operator image envvar not specified")
+		}
+
+		cfg := runtime.Config{
+			Image:          image,
+			EnabledChecks:  []string{enabledCheck},
+			ResponseFormat: DefaultOutputFormat,
+			Mounted:        true,
+		}
+
+		engine, err := engine.NewForConfig(cfg)
+		if err != nil {
+			return err
+		}
+
+		formatter, err := formatters.NewForConfig(cfg)
+		if err != nil {
+			return err
+		}
+
+		resultsFile, err := os.OpenFile(
+			resultsFilenameWithExtension(formatter.FileExtension()),
+			os.O_WRONLY|os.O_CREATE|os.O_APPEND,
+			0600,
+		)
+
+		if err != nil {
+			return err
+		}
+
+		// // also write to stdout
+		resultsOutputTarget := io.MultiWriter(os.Stdout, resultsFile)
+
+		// execute the checks
+		engine.ExecuteChecks()
+		results := engine.Results()
+
+		// return results to the user and then close output files
+		formattedResults, err := formatter.Format(results)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprint(resultsOutputTarget, string(formattedResults))
+		// if err := resultsFile.Close(); err != nil {
+		// 	return err
+		// }
```

Comment:
If I'm understanding this correctly, this command would effectively run a single command in a specific runtime environment (i.e. the `podman unshare` namespace). From this perspective, we'd have to figure out exactly how we handle the outputs.

I may be misunderstanding, but it feels like we should be capturing the output of this internal execution and _merging_ it with the rest of our output. I may be misunderstanding the complete picture here, so any guidance would help!

Reply:
You are correct. I need to figure out the results processing.